### ISSUE TITLE
Response object of `task.update` seems to fail though actual update succeeds

### ISSUE THREAD

**Issue by spadachris** [Asana/ruby-asana#68]
When I run the following code to update a given task, the response is a `Asana::Resources::Task` object, but some functions on that object fail (`projects`, `tags`):

```
task = Asana::Client.new(...).tasks.find_by_id "asana_task_id"
response = task.update(completed: true)
``` 

In the rails console, `response` results in:
```
(pry) output error: #<NoMethodError: undefined method `get' for nil:NilClass
Did you mean?  gem>
```
and `response.to_s` results in:
```
NoMethodError: undefined method `get' for nil:NilClass
Did you mean?  gem
from /app/vendor/bundle/ruby/2.4.0/gems/asana-0.8.1/lib/asana/resources/task.rb:316:in `projects'
```

Looking at `task.rb:316` it looks like `client` is `nil`

Note that the update succeeds and I see the tasked marked as `complete` on asana.com.

**Comment by rossgrambo** [comment]
We plan to do an update of the client libraries within the next few weeks. I'll ensure this issue is solved as a part of that deployment.

**Comment by rossgrambo** [comment]
Hey @spadachris, this fix is live in 0.9.2.

There's another issue to address, which is when you run update, the resulting object is a different object than the one you sent. (but this isn't a blocker, and could break some apps)

But this issue was caused by that duplicate object not being passed the asana client correctly:

```
# objectWithClient >> Completed = false, client = Asana::Client

response = objectWithClient.update (completed = true)

# objectWithClient >> Completed = false, client = Asana::Client
# response >> Completed = true, client = nil
```

Now response has `client = Asana::Client`, so a nice way to use `update` is:

```
objectWithClient = objectWithClient.update(completed = true)
```
